server/src/printers/NativeMakefilePrinter.h

```diff
@@ -15,8 +15,7 @@ namespace printer {
     class NativeMakefilePrinter : public RelativeMakefilePrinter {
         friend class TestMakefilesPrinter;
     private:
-        const utbot::ProjectContext projectContext;
-        std::shared_ptr<BuildDatabase> buildDatabase;
+        const BaseTestGen* testGen;
```

Comment:
Consider making it `const BaseTestGen const * testGen;` to prevent the field from assignment